src/App.tsx

```diff
@@ -1,24 +1,258 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import {
+  FC,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import { addTodo, getTodos, removeTodo } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Loader } from './components/Loader';
+import { FilterBy } from './types/FilteredBy';
+import { TodoFilter } from './components/TodoFilter';
+import { Form } from './components/Form';
+import { ErrorMessage } from './components/ErrorMessage';
 
-const USER_ID = 0;
+const USER_ID = 6894;
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+function filterTodosByCompleted(todos: Todo[], filterBy: string): Todo[] {
+  switch (filterBy) {
+    case FilterBy.ACTIVE:
+      return todos.filter(todo => !todo.completed);
+
+    case FilterBy.COMPLETED:
+      return todos.filter(todo => todo.completed);
+
+    default:
+      return todos;
   }
+}
+
+export interface TodoItem {
+  todo: Todo,
+  isLoading: boolean,
+}
+
+export const App: FC = () => {
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [todoTitle, setTodoTitle] = useState('');
+  const [filterType, setFilterType] = useState<FilterBy>(FilterBy.ALL);
+  const [tempTodos, setTempTodos] = useState<Todo | null>(null);
+  const [activeInput, setActiveInput] = useState(true);
+  const [isQuery, setIsQuery] = useState(false);
+  const [deletedTodoIds, setDeletedTodoIds] = useState<number[]>([]);
+
+  const addError = (message: string) => {
+    setErrorMessage(message);
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  const loadAllTodos = async () => {
+    try {
+      const todos = await getTodos(USER_ID);
+
+      setAllTodos(todos);
+      setIsQuery(true);
+    } catch {
+      addError('Unable to update a todo');
+    } finally {
+      setTempTodos(null);
+      setActiveInput(true);
+    }
+  };
+
+  const handleRemoveTodo = useCallback(
+    (id: number) => {
+      setDeletedTodoIds(prev => [...prev, id]);
+
+      setTimeout(() => {
+        try {
+          removeTodo(id)
+            .then(() => {
+              const todosWithoutDeleted = allTodos.filter((todo) => (
+                todo.id !== id
+              ));
+
+              setAllTodos(todosWithoutDeleted);
+            });
+        } catch {
+          setErrorMessage('Unable to delete a todo');
```

Comment:
```suggestion
          addError('Unable to delete a todo');
```

Reply:
I tried to simulate delete error and I didnt saw it. Can you check it and correct me if i'm wrong?